internal/clients/http/ec2/ec2_client.go

```diff
@@ -28,6 +30,7 @@ const TraceName = "github.com/RHEnVision/provisioning-backend/internal/clients/h
 type ec2Client struct {
 	ec2     *ec2.Client
 	sts     *sts.Client
+	iam     *iam.Client
 	assumed bool
 }
```

Comment:
This seems unused, would you mind to continue and add the next step in this PR?

Reply:
Sure, I had a follow-up prepared, but I was not sure if this would not be too much.